gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -841,37 +841,38 @@ function TellTraitorsAboutTraitors()
 		end
 	end
 
-	hook.Run("TTT2TellTraitors", traitornicks)
+	for _, v in ipairs(player.GetAll()) do
+		if v:HasTeam(TEAM_TRAITOR) then
 
-	-- This is ugly as hell, but it's kinda nice to filter out the names of the
-	-- traitors themselves in the messages to them
-	local traitornicks_min = #traitornicks < 2
+			local tmp = table.Copy(traitornicks)
 
-	if not traitornicks_min then
-		table.sort(traitornicks, function(a, b)
-			return a and b and a:upper() < b:upper()
-		end)
-	end
+			local shouldShow = hook.Run("TTT2TellTraitors", tmp, v)
 
-	for _, v in ipairs(player.GetAll()) do
-		if v:HasTeam(TEAM_TRAITOR) then
-			if traitornicks_min then
-				LANG.Msg(v, "round_traitors_one")
+			if shouldShow == false or tmp == nil or #tmp == 0 then continue end
 
+			if #tmp == 1 then
+				LANG.Msg(v, "round_traitors_one")
 				return
-			else
-				local names = ""
+			end
 
-				for _, name in ipairs(traitornicks) do
-					if name ~= v:Nick() then
-						names = names .. name .. ", "
-					end
-				end
+			if #tmp >= 3 then
```

Comment:
last little thing: `if #tmp > 1 then` :D

Reply:
why already sort it if only 2 names (where one is my own) is in?

Reply:
ah, you are right, forgot the own one that is removed later